Emby.Server.Implementations/IO/ManagedFileSystem.cs

```diff
@@ -149,6 +149,26 @@ public virtual void CreateShortcut(string shortcutPath, string target)
             }
         }
 
+        /// <inheritdoc />
+        public void MoveDirectory(string source, string destination)
+        {
+            try
+            {
+                Directory.Move(source, destination);
+            }
+            catch (IOException)
+            {
+                // Cross device move requires a copy
+                Directory.CreateDirectory(destination);
+                foreach (string file in Directory.GetFiles(source))
+                {
+                    File.Copy(file, Path.Combine(destination, Path.GetFileName(file)), true);
```

Comment:
I think we need to handle errors better here because various common user errors could happen. For example:

- The destination drive is full and only part of the files are moved (a common user error).
- The destination path is too long (common with network mounts).
- A trickplay file already existing in the destination folder (also common as some users share media between multiple instances).

Some of these issues, such as file already exists, should not be considered failures, and we can just delete the source folder. The problem here is that both "file exists" and "destination drive is full" will throw an `IOException`. If the performance allows, maybe we should checking if the file exists before performing the copy?

Reply:
I don't think we should be the ones responsible for handling `drive is full` but I do agree that the other two cases should be handled better. Will look into it.

Reply:
Do we handle destination is ro? This is also common.

Reply:
This will fail with a permission error and we will bail out. Which is totally fine to me as we can do nothing anyway.

Reply:
Just realized we are calling copy with `overrde = true` so case C is already handled.